src/ripple/rpc/tests/JSONRPC.test.cpp

```diff
@@ -784,9 +850,89 @@ R"({
 {
 "Secret does not match account.",
 "Secret does not match account.",
-"Missing field 'tx_json.SigningPubKey'.",
+"",
+"Missing field 'tx_json.SigningPubKey'."}},
+
+{ "In sign_for, and account may not sign for itself.",
+R"({
+    "command": "doesnt_matter",
+    "account": "rnUy2SHTrB9DubsPmkJZUXTf5FcNDGrYEA",
+    "secret": "a",
+    "tx_json": {
+        "Account": "rnUy2SHTrB9DubsPmkJZUXTf5FcNDGrYEA",
+        "Amount": "1000000000",
+        "Destination": "rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh",
+        "Fee": 50,
+        "Sequence": 0,
+        "TransactionType": "Payment"
+    }
+})",
+{
+"",
+"",
+"A Signer may not be the transaction's Account (rnUy2SHTrB9DubsPmkJZUXTf5FcNDGrYEA).",
 "Missing field 'tx_json.SigningPubKey'."}},
 
+{ "Cannot put duplicate accounts in Signers array",
+R"({
+    "command": "doesnt_matter",
+    "account": "rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh",
+    "secret": "masterpassphrase",
+    "tx_json": {
+        "Account" : "rnUy2SHTrB9DubsPmkJZUXTf5FcNDGrYEA",
+        "Amount" : "1000000000",
+        "Destination" : "rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh",
+        "Fee" : "50",
+        "Sequence" : 0,
+        "Signers" : [
+            {
+                "Signer" : {
+                    "Account" : "rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh",
+                    "SigningPubKey" : "0330E7FC9D56BB25D6893BA3F317AE5BCF33B3291BD63DB32654A313222F7FD020",
+                    "TxnSignature" : "304502210080EB23E78A841DDC5E3A4F10DE6EAF052207D6B519BF8954467ADB221B3F349002202CA458E8D4E4DE7176D27A91628545E7B295A5DFC8ADF0B5CD3E279B6FA02998"
+                }
+            }
+        ],
+        "SigningPubKey" : "",
+        "TransactionType" : "Payment"
+    }
+})",
+{
+"Secret does not match account.",
+"Secret does not match account.",
+"Duplicate Signers:Signer:Account entries (rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh) are not allowed.",
+""}},
+
+{ "Correctly append to pre-established Signers array",
+R"({
+    "command": "doesnt_matter",
+    "account": "rPcNzota6B8YBokhYtcTNqQVCngtbnWfux",
```

Comment:
Yeah, it is useful context.  When I first started building these test cases there were strings like "submit" in there.  But I send each of these test cases through all four of `sign`, `sign_for`, `submit`, and `submit_multisigned`.  So I felt like leaving text like "submit" or "sign" was misleading.  Given that, I felt like "doesn't matter" was a reasonable choice.